opus/application/static_media/js/browse.js

```diff
@@ -666,47 +666,52 @@ var o_browse = {
         let topBoxBoundary; // assign value in the each loop below to avoid getting type error in views other than #browse and #cart
         let startObsLabel = o_browse.getStartObsLabel();
 
-        $(selector).each(function(index, elem) {
-            // Fot gallery view, the topBoxBoundary is the top of .gallery-contents
-            // For table view, we will set the topBoxBoundary to be the bottom of thead (account for height of thead)
+        if ($(selector).length > 0) {
+            // this will get the top left obsNum for gallery view or the top obsNum for table view
+            let firstCachedObs = $(selector).first().data("obs");
+            let firstCachedObsTop = $(selector).first().offset().top;
             topBoxBoundary = topBoxBoundary || (opus.prefs.browse === "dataTable") ? $(`${tab} .gallery-contents`).offset().top + $(`${tab} #dataTable thead th`).outerHeight() : $(`${tab} .gallery-contents`).offset().top;
 
-            // compare the image .top + half its height in order to make sure we account for partial images
-            let topBox = $(elem).offset().top + $(elem).height()/2;
-            if (topBox >= topBoxBoundary) {
-                let obsNum = $(elem).data("obs");
+            // table: obsNum = firstCachedObs + number of row
+            // gallery: obsNum = firstCachedObs + number of row * number of obs in a row
+            let obsNumDiff = (opus.prefs.browse === "dataTable") ? Math.round((topBoxBoundary - firstCachedObsTop)/$(`${tab} tbody tr`).outerHeight()) : Math.round((topBoxBoundary - firstCachedObsTop)/o_browse.imageSize) * o_browse.galleryBoundingRect.x;
```

Comment:
Break into multiple lines so it's more readable

Reply:
Also, this won't work for the cart, will it? opus.prefs.browse is only for the main browse tab. Need to use Debby's new functions that retrieve the proper browse name.